packages/tenant-readmodel-writer/src/tenantConsumerServiceV2.ts

```diff
@@ -0,0 +1,48 @@
+import { TenantCollection } from "pagopa-interop-commons";
+import { TenantEventEnvelopeV2, fromTenantV2 } from "pagopa-interop-models";
+import { match } from "ts-pattern";
+
+export async function handleMessageV2(
+  message: TenantEventEnvelopeV2,
+  tenants: TenantCollection
+): Promise<void> {
+  const tenant = message.data.tenant;
+
+  await match(message)
+    .with({ type: "MaintenanceTenantDeleted" }, async (message) => {
+      await tenants.deleteOne({
+        "data.id": message.stream_id,
+        "metadata.version": { $lt: message.version },
+      });
+    })
+    .with(
+      { type: "TenantOnboarded" },
+      { type: "TenantOnboardDetailsUpdated" },
+      { type: "TenantCertifiedAttributeAssigned" },
+      { type: "TenantCertifiedAttributeRevoked" },
+      { type: "TenantDeclaredAttributeAssigned" },
+      { type: "TenantDeclaredAttributeRevoked" },
+      { type: "TenantVerifiedAttributeAssigned" },
+      { type: "TenantVerifiedAttributeRevoked" },
+      { type: "TenantVerifiedAttributeExpirationUpdated" },
+      { type: "TenantVerifiedAttributeExtensionUpdated" },
+      { type: "TenantMailAdded" },
+      async (message) =>
+        await tenants.updateOne(
+          {
+            "data.id": message.stream_id,
+            "metadata.version": { $lt: message.version },
+          },
+          {
+            $set: {
+              data: tenant ? fromTenantV2(tenant) : undefined,
+              metadata: {
+                version: message.version,
+              },
+            },
+          },
+          { upsert: true }
+        )
+    )
+    .exhaustive();
```

Comment:
missing events
- `MaintenanceTenantPromotedToCertifier`
- `TenantMailDeleted`

can you check exhaustive match here as well?

Reply:
At the beginning I seem to remember that it was not known whether these 2 events would be used or not, which is why they were added later with this PR

- MaintenanceTenantPromotedToCertifier https://github.com/pagopa/interop-be-monorepo/pull/462

- TenantMailDeleted https://github.com/pagopa/interop-be-monorepo/pull/461